### DISCUSSION TITLE
Question: CSS to Xamarin.Forms

### DISCUSSION THREAD

**Discussion by juucustodio** [dotnet/maui#44] [General]
The Xamarin.Forms 3.0 and latest versions have support to CSS. 
How the situation about this ? MAUI will support this feature ?

**Comment by charlesroddie** [comment]
I think it's best to remove this. It is additional code to maintain, additional docs to maintain, confusion for users, and extra work needed to add or change controls in the repository.

For interested users this function could be in a separate community-maintained repository, usable via a separate nuget package. CSharp source generators would allow this to happen without any need to provide support in the main repo.

It's important to keep the main repo as small and clean as possible for maintainability.

**Comment by GalaxiaGuy** [comment]
I much prefer XAML styling, but there are currently things you can do with CSS you can't do with XAML styling.

Whether CSS is dropped or not, I would like XAML styling to support advanced selectors.

**Reply by Lifehack25** [reply]
Yeah CSS is very established and there are lots of cool pre-coded animations and designs to get from the web. Much more than XAML design.

**Comment by ghost** [comment]
Genuinely curious about how many people use CSS styles over XAML styles? I know this is purely anecdotal, but I’ve never seen it used outside of examples.

I’d agree with @charlesroddie, add support via NuGet for people who want it.

**Comment by juucustodio** [comment]
> Genuinely curious about how many people use CSS styles over XAML styles? I know this is purely anecdotal, but I’ve never seen it used outside of examples.
> 
> I’d agree with @charlesroddie, add support via NuGet for people who want it.

I agree with @charlesroddie opinion.  Maybe is a solution of this case. 

Responding about your question... I know people that use CSS to Xamarin.Forms. But the idea not is choice between XAML Styles and CSS. The people in this case use together.

**Comment by jonedmiston** [comment]
Just putting in a vote to keep CSS styling in the core. We use it exclusively for styling in our framework and love the flexibility and easy it provides.

**Comment by felipebaltazar** [comment]
maybe using the generic [host structure (issue spec)](https://github.com/dotnet/maui/issues/24) we can split these "styles features" to optional modules (nuget).
So, the developer can configure one, or both, on `startup.cs`

something like...

```c#
Host.CreateDefaultBuilder()
    .ConfigureStyles(settings => settings.UseXamlStyles().UseCssStyles())
```

Where `UseXamlStyles` and `UseCssStyles` are extension methods from nuget packages

**Comment by GiampaoloGabba** [comment]
To me, is the linker that should be in charge to strip away the css assemblies if not used in app :) 
I think that this is how it works today.

I'm pretty sure css will not be removed from MAUI, in some cases you can do some things not possible with xaml styling.

**Comment by felipebaltazar** [comment]
> To me is the linker that should be in charge to strip away the CSS assemblies if not used in-app :)
I think that this is how it works today.

This is a good point!
In my opinion, Linker is an advanced tool that some developers don't use properly...
Splitting some components can be a way to reduce app sizes for these "novice developers".
Btw, for me, it makes no sense to include code to remove it later.
Off course, my suggestion will increase assemblies to load, this will be a problem

But I don't know, maybe I'm just talking about something that has already been discussed 🤷‍♂️

**Comment by pictos** [comment]
I think that makes sense to maintain the CSS support **if** we have web support, as described here #62. Also, if the **Mobile Blazor Bindings** becomes a thing, the style using CSS will be more useful than XAML style (in this case).

**Reply by MihirJayavant** [reply]
CSS support is important for **Mobile Blazor Bindings**. It would be bad idea to use XAMLstyles there
